cmd/notify-mailer/main.go

```diff
@@ -130,68 +137,68 @@ func (m *mailer) run() error {
 
 	startTime := m.clk.Now()
 
-	for i, dest := range destinations {
-		m.printStatus(dest, i+1, len(destinations), startTime)
-		if strings.TrimSpace(dest) == "" {
+	sortedAddresses := sortAddresses(addressesToRecipients)
+
+	var sent int
+	for i, address := range sortedAddresses {
+		if !m.targetRange.includes(address) {
+			m.log.Debugf("skipping %q: out of target range")
 			continue
 		}
-		err := m.mailer.SendMail([]string{dest}, m.subject, m.emailTemplate)
+		recipients := addressesToRecipients[address]
+		i += 1
+		m.printStatus(address, i, len(addressesToRecipients), startTime)
+		var mailBody bytes.Buffer
+		err = m.emailTemplate.Execute(&mailBody, recipients)
 		if err != nil {
 			switch err.(type) {
 			case bmail.InvalidRcptError:
-				m.log.Errf("address %q was rejected by server: %s", dest, err)
+				m.log.Errf("address %q was rejected by server: %s", address, err)
 				continue
 			default:
 				return err
 			}
 		}
+		if mailBody.Len() == 0 {
+			return fmt.Errorf("email body was empty after interpolation.")
+		}
+		err := m.mailer.SendMail([]string{address}, m.subject, mailBody.String())
+		if err != nil {
+			return err
+		}
+		sent++
 		m.clk.Sleep(m.sleepInterval)
 	}
+	if sent == 0 {
+		return fmt.Errorf("skipped all addresses. bad interval?")
```

Comment:
I think this error would be clearer if it referenced the `sent` variable beyond mentioning addresses.

```suggestion
		return fmt.Errorf("sent zero messages. Check recipients and configured interval")
```